util/stmtsummary/statement_summary.go

```diff
@@ -292,6 +302,25 @@ func (ssMap *stmtSummaryByDigestMap) Clear() {
 	ssMap.beginTimeForCurInterval = 0
 }
 
+// ClearInternal removes all statement summaries which are internal summaries.
+func (ssMap *stmtSummaryByDigestMap) ClearInternal() {
+	ssMap.Lock()
+	defer ssMap.Unlock()
+
+	for _, value := range ssMap.summaryMap.Values() {
+		summary := value.(*stmtSummaryByDigest)
+		for ele := summary.history.Front(); ele != summary.history.Back(); {
+			if ele.Value.(*stmtSummaryByDigestElement).isInternal {
+				delElement := ele
+				ele = delElement.Next()
+				summary.history.Remove(delElement)
+			} else {
+				ele = ele.Next()
+			}
+		}
+	}
```

Comment:
Only history is removed, the whole summary is not removed, and it still occupies the capacity for the LRU cache.